test/ParameterStoreBuildWrapperPluginTest.groovy

```diff
@@ -25,6 +33,154 @@ class ParameterStoreBuildWrapperPluginTest {
             Collection actualPlugins = TerraformEnvironmentStage.getPlugins()
             assertThat(actualPlugins, hasItem(instanceOf(ParameterStoreBuildWrapperPlugin.class)))
         }
+
+        @Test
+        void modifiesTerraformValidateStageCommand() {
+            ParameterStoreBuildWrapperPlugin.init()
+
+            Collection actualPlugins = TerraformValidateStage.getPlugins()
+            assertThat(actualPlugins, hasItem(instanceOf(ParameterStoreBuildWrapperPlugin.class)))
+        }
+    }
+
+    public class Apply {
```

Comment:
The idea behind the classes is to logically group tests.  Junit doesn't let you group tests, so  using classes is a kind of work around.  As a convention, I try to name each grouping/class after the method that I want to test. 

In your case, there are actually 2 `apply` methods.  One that works with TerraformValidateStage, and one that works with TerraformEnvironmentStage.  They're different methods  with different behaviors.  I would consider splitting them out.  Using the TagPlugin  as an example, you could do an "ApplyForTerraformValidateStage", and an "ApplyForTerraformEnvironmentStage".  Or you could even do nested:
```
...
    class Apply {
        class WithTerraformValidateStage {
             ....
        }

        class  WithTerraformEnvironmentStage {
            ....
        }
    }
...
```

Reply:
opted for the nested route